datacube/drivers/postgis/_api.py

```diff
@@ -651,6 +652,20 @@ def search_datasets_query(self,
         if not select_fields:
             select_fields = _dataset_fields()
 
+        def _ob_exprs(o):
+            if isinstance(o, str):
+                # does this need to be more robust?
+                return text(o)
```

Comment:
A string here should be the name of a known/system field only, in which case, convert to a field and proceed.